File: CPP/Aksesoris.cpp
/*Saya Klara Ollivviera Augustine Gunawan dengan NIM 2306205 
mengerjakan soal Tugas Praktikum 2 dalam mata kuliah DPBO 
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin */
#include <iostream>
#include <string>
#include "Petshop.cpp"

using namespace std;

class Aksesoris : public Petshop{//inheritance kelas Aksesoris kepada kelas Petshop
    private://atribut kelas Aksesoris
        string jenis;
        string bahan;
        string warna;

    public:
        Aksesoris(){//konstruktor
            this->jenis = "";
            this->bahan = "";
            this->warna = "";
        }

        Aksesoris(string id, string nama_produk, int harga_produk, int stok_produk, string jenis, string bahan, string warna) : Petshop(id, nama_produk, harga_produk, stok_produk){
            this->jenis = jenis;
            this->bahan = bahan; 
            this->warna = warna;
        }

        //getter dan setter
        string get_jenis(){
            return this->jenis;
        }

        void set_jenis(string jenis){
            this->jenis = jenis;
        }

        string get_bahan(){
            return this->bahan;
        }

        void set_bahan(string bahan){
            this->bahan = bahan;
        }

        string get_warna(){
            return this->warna;
        }

        void set_warna(string warna){
            this->warna = warna;
        }

        ~Aksesoris(){//destruktor

        }
};

File: PHP/Aksesoris.php
<!-- Saya Klara Ollivviera Augustine Gunawan dengan NIM 2306205 
mengerjakan soal Tugas Praktikum 1 dalam mata kuliah DPBO 
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin -->

<?php

include 'Petshop.php';

class Aksesoris extends Petshop{
    private $jenis = "";
    private $bahan = "";
    private $warna = "";

    public function __construct($id = "", $nama_produk = "", $harga_produk = 0, $stok_produk = 0, $foto_produk = "", $jenis = "", $bahan = "", $warna = ""){
        parent::__construct($id, $nama_produk, $harga_produk, $stok_produk, $foto_produk);
        $this->jenis = $jenis;
        $this->bahan = $bahan;
        $this->warna = $warna;
    }

    public function set_jenis($jenis){
        $this->jenis = $jenis;
    }

    public function get_jenis(){
        return $this->jenis;
    }

    public function set_bahan($bahan){
        $this->bahan = $bahan;
    }

    public function get_bahan(){
        return $this->bahan;
    }

    public function set_warna($warna){
        $this->warna = $warna;
    }

    public function get_warna(){
        return $this->warna;
    }
}
?>